lib/components/user/monitored-trip/trip-status.js

```diff
@@ -0,0 +1,391 @@
+   * assumed that the journeyState of the monitored trip will be either
+   * `TRIP_UPCOMING` or `TRIP_ACTIVE`.
+   */
+  renderRealtimeInfo = () => {
+    const { monitoredTrip } = this.props
+    const journeyState = this.getJourneyState()
+
+    const { dateFormat, timeFormat } = this.props
+    const extendedDateTimeFormat = `dddd ${dateFormat} [at] ${timeFormat}`
+    const { matchingItinerary } = journeyState
+    const tripStartMoment = moment(matchingItinerary.startTime)
+    const tripEndMoment = moment(matchingItinerary.endTime)
+
+    let body
+    let headerText
+    let panelBsStyle = 'info'
+    if (journeyState.tripStatus === 'TRIP_UPCOMING') {
+      const minutesUntilTripStart = getMinutesUntilItineraryStart(
+        matchingItinerary
+      )
+      if (minutesUntilTripStart > monitoredTrip.leadTimeInMinutes) {
+        headerText = `Next trip starts on ${tripStartMoment.format(extendedDateTimeFormat)}.`
+      } else {
+        // analyze whether the journey state indicates that the matching
+        // itinerary has realtime data. It is assumed that realtime data will
+        // exist for itineraries that have differing values of the scheduled
+        // departure time and matching itinerary departure time.
+        if (journeyState.hasRealtimeData) {
+          // calculate the deviation from the scheduled departure time (positive
+          // value indicates delay)
+          const departureDeviationSeconds = (
+            matchingItinerary.startTime -
+            journeyState.scheduledDepartureTimeEpochMillis
+          ) / 1000
+          const deviationHumanDuration = formatDuration(
+            Math.abs(departureDeviationSeconds)
+          )
+          if (Math.abs(departureDeviationSeconds) < ON_TIME_THRESHOLD_SECONDS) {
+            // about on time
+            headerText = 'Trip is starting soon and is about on time.'
+            panelBsStyle = 'success'
+          } else if (departureDeviationSeconds > 0) {
+            // delayed
+            headerText = `Trip start time is delayed ${deviationHumanDuration}!`
+            panelBsStyle = 'warning'
+          } else {
+            // early
+            headerText = `Trip start time is happening ${deviationHumanDuration} earlier than expected!`
+            panelBsStyle = 'warning'
+          }
+        } else {
+          headerText = 'Trip is starting soon (no realtime updates available).'
+          panelBsStyle = undefined
+        }
+      }
+
+      body = `Trip is due to begin at ${tripStartMoment.format(timeFormat)}.`
+    } else if (journeyState.tripStatus === 'TRIP_ACTIVE') {
+      // analyze whether the journey state indicates that the matching itinerary
+      // has realtime data. It is assumed that realtime data will exist for
+      // itineraries that have differing values of the scheduled arrival time
+      // and matching itinerary arrival time.
+      if (journeyState.hasRealtimeData) {
```

Comment:
The conditions the block below are similar to the if block above, but I don't see a good way to refactor.